routing/turns_sound_settings.cpp

```diff
@@ -73,7 +72,23 @@ double Settings::ConvertUnitsToMeters(double distanceInUnits) const
       return MeasurementUtils::FeetToMeters(distanceInUnits);
   }
 
-  // m_lengthUnits is equal to LengthUnits::Undefined or to unknown value.
+  ASSERT(false, ());
+  return 0.;
+}
+
+double Settings::ConvertMetersToUnits(double distanceInMeters) const
+{
+  switch (m_lengthUnits)
+  {
+    case LengthUnits::Undefined:
```

Comment:
Is it hard to refactor here?
Is it the only place where Undefined is used?

> On Oct 9, 2015, at 01:48, Vladimir Byko-Ianko <notifications@github.com> wrote:
> 
> In routing/turns_sound_settings.cpp <https://github.com/mapsme/omim/pull/168#discussion_r41609187>:
> 
> > @@ -73,7 +72,23 @@ double Settings::ConvertUnitsToMeters(double distanceInUnits) const
> >        return MeasurementUtils::FeetToMeters(distanceInUnits);
> >    }
> >  
> > -  // m_lengthUnits is equal to LengthUnits::Undefined or to unknown value.
> > +  ASSERT(false, ());
> > +  return 0.;
> > +}
> > +
> > +double Settings::ConvertMetersToUnits(double distanceInMeters) const
> > +{
> > +  switch (m_lengthUnits)
> > +  {
> > +    case LengthUnits::Undefined:
> Yes, you're right. I needed Undefined before but not now.
> 
> I guess we need:
> 1. to remove Undefined form LengthUnits;
> 2. to move LengthUnits to platform/settings.hpp and to remove Unit enum there;
> 3. Use enum class LengthUnits instead of enum Unit;
> 
> But I think it's a question of separate PR.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/168/files#r41609187>.
>